### DISCUSSION TITLE
How to specify cell parameters for languages with unusual comments

### DISCUSSION THREAD

**Discussion by bvancil** [quarto-dev/quarto-cli#3876] [Q&A] [answered]
Greetings! I'm happy to try to put together an alternative minimal reproducible example in another language if someone can suggest a Jupyter kernel that uses less common comments. I'm trying to label a table with a caption in this example with SAS, which uses an asterisk (*) to begin the comment and a semicolon (;) to end it. It also has C-style /* ... */ comments.

````
---
format:
    html:
        code-fold: true
jupyter: sas
---

As you can see in Table @table-class, there are people.

```{sas}
*| label: table-class;
*| fig-cap: "Class table";
title;
proc print data = sashelp.class(obs = 4);
run;
```
````

results in 

```
YAMLException: bad indentation of a mapping entry (src\quarto\prototype\test_table-caption.qmd, 12:11)
11: *| label: table-class;
12: *| fig-cap: "Class table";
             ~~
13: title;
```

Getting rid of the figure caption produces output, but the label reference isn't resolved, so I'm clearly specifying the parameters wrongly. Any ideas for how to specify parameters in a way that both SAS and Quarto will like?

### Things I've tried

```
*| label: table-class
*| fig-cap: "Class table"
;
```

Result: Quarto doesn't complain, but neither the caption nor the reference work.

```
/*
*| label: table-class
*| fig-cap: "Class table"
*/
```

Result: Same

```
/*
#| label: table-class
#| fig-cap: "Class table"
*/
```

Result: Same

**Comment by cscheid** [comment, marked as answer]
This feels like a bug on our side (SAS is one of the few languages we're trying to support which has "open" and "close" comment syntax). I'll open an issue to track.

(Incidentally, table id syntax is `tbl-`, not `table-`)

**Reply by bvancil** [reply]
Is there an open-source language with an open/close syntax? If so, I'm happy to write a test in that language to make it easier to test.

**Reply by cscheid** [reply]
I think the only one that technically applies is `c`, and you'd have to install the Jupyter C kernel (which I don't know how to do, sorry): https://github.com/quarto-dev/quarto-cli/blob/main/src/core/jupyter/jupyter.ts#L957-L1000

If you figure this out, though, I'm happy to learn how to do it so we can fix the bug.
